src/main/java/com/conveyal/r5/analyst/scenario/IndexedPolygonCollection.java

```diff
@@ -104,9 +107,11 @@ public IndexedPolygonCollection (
     }
 
     public void loadFromS3GeoJson() throws Exception {
-        InputStream polygonInputStream = WorkerComponents.fileStorage.getInputStream(POLYGONS, polygonLayer);
-        FeatureJSON featureJSON = new FeatureJSON();
-        FeatureCollection featureCollection = featureJSON.readFeatureCollection(polygonInputStream);
+        // FIXME this needs to be adapted to new SpatialDataSource. How will we handle .gz data?
```

Comment:
Are there end users who rely on .gzipped congestion area polygons?

Reply:
We do have some pre-existing gzipped congestion areas already on S3. However they're in a different bucket (polygons) and this has been explained to users as a new feature subject to change. So if we get gzip support added in time, we still need to move all those polygon files over to the datasources bucket on S3 and add database records for them. Probably the easiest way to do that is to just re-upload them via the UI. If we re-up them without gzipping them, I believe most of the layers involved (client to backend, backend to s3) will automatically compress the data stream so it's not that wasteful - only the storage space on s3 will be larger. Gzip support is only tricky because it alters the recognizable extensions of the files, and is typically implemented as an InputStream wrapping the FileInputStream, but GeoTools reads from Files rather than InputStreams.

I kind of suspect we're going to eventually end up implementing our own GeoJson support that works in a completely streaming manner, to allow for gzipped files and very large files, and to validate the five or six important constraints we're counting on when reading GeoJson. But for the time being we will be leaning on GeoTools so have to work within the limitations of its GeoJson extension.